python/ray/dataframe/dataframe.py

```diff
@@ -2774,37 +2773,66 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
                     are the quantiles.
         """
 
-        def quantile_helper(df, q, axis, numeric_only, interpolation):
-            try:
-                return df.quantile(q=q, axis=axis, numeric_only=numeric_only,
-                                   interpolation=interpolation)
-            except ValueError:
-                return pd.Series()
+        # deal with error checking for bad dtyping when numeric_only is False
+        if not numeric_only:
+            # check if there are any object columns
+            for t in self.dtypes:
+                if t == np.dtype('O'):
+                    e_msg = "Cannot call quantile on columns of type " + str(t)
+                    raise TypeError(e_msg)
+
+        # check that all qs are between 0 and 1
+        msg = ("percentiles should all be in the interval [0, 1]. "
+               "Try {0} instead.")
 
         if isinstance(q, (pd.Series, np.ndarray, pd.Index, list)):
-            # In the case of a list, we build it one at a time.
-            # TODO Revisit for performance
-            quantiles = []
-            for q_i in q:
-                def remote_func(df):
-                    return quantile_helper(df, q=q_i, axis=axis,
-                                           numeric_only=numeric_only,
-                                           interpolation=interpolation)
-
-                result = self._arithmetic_helper(remote_func, axis)
-                result.name = q_i
-                quantiles.append(result)
-
-            return pd.concat(quantiles, axis=1).T
+            if not all(0 <= qs <= 1 for qs in q):
+                raise ValueError(msg.format(np.asarray(q) / 100.0))
         else:
+            if not (0 <= q <= 1):
+                raise ValueError(msg.format(q / 100.0))
+
+        if isinstance(q, (pd.Series, np.ndarray, pd.Index, list)):
```

Comment:
Use `axis = pd.DataFrame()._get_axis_number(axis)` instead.

Reply:
What would this do here? I'm checking if q is a list or not